tests/perf-system/analyzer/analyzer.py

```diff
@@ -0,0 +1,290 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import pandas as pd  # type: ignore
+
+# pylint: disable=import-error
+from prettytable import PrettyTable  # type: ignore
+import numpy as np
+import matplotlib.pyplot as plt  # type: ignore
+from typing import List
+import sys
+from loguru import logger as LOG
+
+
+SEC_MS = 1000
+
+# Change default log format
+LOG.remove()
+LOG.add(
+    sys.stdout,
+    format="<green>[{time:HH:mm:ss.SSS}]</green> {message}",
+)
+
+
+class Analyze:
+    def __init__(self):
+        self.latency_list = []
+        self.ms_latency_list = []
+
+    def get_req_type(self, df_responses: pd.DataFrame) -> str:
+        return df_responses.iloc[0]["rawResponse"].split(" ")[0]
+
+    def get_latency_at_i(
+        self, df_sends: pd.DataFrame, df_responses: pd.DataFrame, req_id: int
+    ):
+        # will need to handle the re-submission response from submitter when decided
+        return (
+            df_responses.iloc[req_id]["receiveTime"] - df_sends.iloc[req_id]["sendTime"]
+        )
+
+    def check_success(self, df_responses: pd.DataFrame, req_id: int) -> int:
+        req_resp = df_responses.iloc[req_id]["rawResponse"].split("\n")
+        status_list = req_resp[0].split(" ")
+        # if we get a full statues and says ok increase the successful
+        if (
+            len(status_list) > 1
+            and status_list[1][:3] >= "200"

```

Comment:
This probably mostly works in practice, but doesn't seem ideal, since "3" for example would be accepted. I would be better to enforce something like "2\d\d" I think.

Reply:
I totally missed that! Thanks for catching it. I could probably add a check for the `len()` of the string, but I think it makes sense to import regex at this point, which I was trying to avoid.

Reply:
Much simpler than importing regex is to try and parse this as an int, and confirm that int is between 200 and 299. If its not parseable, or not in the correct range, that's an error.

Reply:
I can't resist pointing out that this comes in as a string, so while you will need an extra `import re`, you can use a single `re.compile("^2\d\d$").match()` as a condition. The conversion to int on the other hand may throw, and that's at least another two lines to handle.

Having said that, I am of course fine with anything that's correct ^_^.

Reply:
I used re which is builtin `re.search("^2[0-9][0-9]$", status_list[1])`. I was thinking as well the parsing @eddyashton but I was wondering if the parsing and comparing would be faster than doing a `re.search()`. I am happy to change it to parsing if you think it is better.

Reply:
`\d` and `[0-9]` produce the same result in our case (if I am not very wrong, their difference is very tiny which would never occur in our case).

Reply:
@fotiskoun https://docs.python.org/3/library/re.html

```
\d

For Unicode (str) patterns:
Matches any Unicode decimal digit (that is, any character in Unicode character category [Nd]). This includes [0-9], and also many other digit characters. If the [ASCII](https://docs.python.org/3/library/re.html#re.ASCII) flag is used only [0-9] is matched.

For 8-bit (bytes) patterns:
Matches any decimal digit; this is equivalent to [0-9].
```